### PR TITLE
Add padding interface to Page content FE-5245

### LINKED ISSUE #5302: Pages content padding can not be removed
### Current behaviour

Pages component have by default some padding that we can not remove it and this limit us a lot on the UI we want to implement

image

### Expected behaviour

The consumer should define the UI , paddings , styles and Pages component should be responsible for the functionality

### CodeSandbox or Storybook URL

https://codesandbox.io/s/fast-snowflake-yfe056?file=/src/index.js

### JIRA Ticket (Sage Only)

SBS-37838

### Suggested Solution

_No response_

### Carbon Version

101.0.2

### Design Tokens Version

_No response_

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Anything else we should know?

_No response_

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

### PR DESCRIPTION
**Pull request by edleeks87**
fix #5302

### Proposed behaviour

Refactors `Pages`, `Page` and `FullScreenHeading` to TypeScript
Adds `styled-system` padding interface to `Page` content container element
to allow consuming projects to override the default padding values.

### Current behaviour

 
`Pages`, `Page` and `FullScreenHeading` are JavaScript
No support for padding interface in `Page` content

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Unit tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

https://codesandbox.io/s/stoic-northcutt-w1eg9k?file=/src/index.js

### CODE REVIEW

**Review by Parsium — DISMISSED** [review]

File: src/components/pages/page/page.spec.tsx [hunk 1]
```diff
@@ -1,13 +1,29 @@
 import React from "react";
 import { shallow } from "enzyme";
-import { StyledPage } from "./page.style.js";
+import { StyledPage, StyledPageContent } from "./page.style";
 import Page from "./page.component";
 import FullScreenHeading from "../../../__internal__/full-screen-heading";
+import { testStyledSystemPadding } from "../../../__spec_helper__/test-utils";
 
 describe("Pages", () => {
```
Comment (Parsium) [inline src/components/pages/page/page.spec.tsx:8]:
```suggestion
describe("Page", () => {
```

File: src/components/pages/pages.spec.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 /* eslint-disable react/prop-types */
 import React, { useState } from "react";
-import { shallow, mount } from "enzyme";
+import { shallow, mount, ReactWrapper, ShallowWrapper } from "enzyme";
 import { act } from "react-dom/test-utils";
 import BasePages, { Page } from "./pages.component";
```
Comment (Parsium) [inline src/components/pages/pages.spec.tsx:5]:
```suggestion
import Pages, { Page } from "./pages.component";
```
What do you think about changing the name to `Pages` like you've done in our stories?

File: src/components/box/box.component.tsx [hunk 3]
```diff
@@ -27,6 +27,8 @@ export interface BoxProps
   overflowWrap?: OverflowWrap;
   /** scroll styling attribute */
   scrollVariant?: ScrollVariant;
+  /** set the box-sizing attribute of the Box component */
+  boxSizing?: "content-box" | "border-box";
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/box/box.component.tsx:31]: Yeah you're right, I checked the styles in the dev console and thought I saw it being applied to the Box but I'm thinking now I was inspecting the wrong element 👏 🤦

**Review by Parsium — APPROVED** [review]

File: src/components/pages/pages.spec.tsx [hunk 4]
```diff
@@ -24,6 +108,8 @@ describe("BasePages", () => {
         </BasePages>
       );
 
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/pages/pages.spec.tsx:111]: Sounds like a reasonable approach 👍

[the remark this answers is not in the document]
Comment (edleeks87) [reply src/components/pages/pages.spec.tsx]: I've just refactored the implementation for now and removed the `ts-ignore`, the effort to get the mock to work for all the tests wasn't worth it imo and can just be tackled when the RTL refactor happens

**Review by mkrds — APPROVED** [review]

**Review by Parsium — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by Parsium — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by Parsium — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Pull request merged**